src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,182 @@
+        # Send grant requests
+        grant_response = self._sas.Grant(request)['grantResponse']
+        grant_ids = []
+        for resp in grant_response:
+            grant_ids.append(resp['grantId'])
+        del request
+        
+        # STEP 2
+        # Inject PPA
+        pal_low_frequency = 3550000000.0
+        pal_high_frequency = 3560000000.0
+        pal_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+        ppa_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+        ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                                [pal_record],
+                                                                pal_low_frequency,
+                                                                pal_high_frequency,
+                                                                device_a['userId'])
+        ppa_record['ppaInfo']['cbsdReferenceId'] = [cbsd_ids[1]]
+        self._sas_admin.InjectPalDatabaseRecord(pal_record[0])
+        self._sas_admin.InjectZoneData({'record': ppa_record})
+        # Inject Esc Sensor
+        esc_sensor_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+        self._sas_admin.InjectEscSensorDataRecord({'record': esc_sensor_record})
+        # STEP 3
+        self._sas_admin.TriggerFullActivityDump()
+        # STEP 4
+        response = self._sas.GetFullActivityDump()
+        # STEP 5
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        generation_date_time = datetime.strptime(response['generationDateTime'],
+                                          '%Y-%m-%dT%H:%M:%SZ')
+        # STEP 6 AND CHECK
+        if generation_date_time >= (datetime.utcnow() - timedelta(minutes=1)):
+            for activity_dump_file in response['files']:
+                # Verify the record type
+                # Verify the response files with ActivityDumpFile.schema.json Object schema
+                self.assertContainsRequiredFields("ActivityDumpFile.schema.json", activity_dump_file)
+                # Get json data from url
+                data = self._sas.DownloadFile(activity_dump_file['url'])
+                if activity_dump_file['recordType'] == 'cbsd':
+                    self.assertEqual(2, len(data['recordData']))
+                    # Verify that cbsds and grants in the full dump matches a cbsd or grant created acitivities
+                    for record in data['recordData']:
+                        # Verify the response files with CbsdData.schema.json Object schema
+                        self.assertContainsRequiredFields("CbsdData.schema.json", record)
+                        self.assertEqual(1, len(record['grants']))
+                        if record['registration']['fccId'] == device_a['fccId']:
```

Comment:
If a SAS returned two records, both with FCC ID == device_c['fccId'], it would pass this part of the test even though it gave the wrong response.

Reply:
I fixed this please feel free to review the new modifications, thank you.